stripe/src/main/java/com/stripe/android/StripeApiHandler.java

```diff
@@ -77,6 +77,21 @@
 
     static final String API_VERSION = "2017-06-05";
 
+    // Used for HTTP-level tests
+    private static @Nullable MockGetStripeResponse mockGetStripeResponse;
+
+    interface MockGetStripeResponse{
+        StripeResponse getStripeResponse(@RestMethod String method,
+                                         String url,
+                                         Map<String, Object> params,
+                                         RequestOptions options);
+    }
+
+    @VisibleForTesting
+    static void setMockGetStripeResponse(MockGetStripeResponse aMockGetStripeResponse){
```

Comment:
I think a better way to go about this would be to create a `StripeResponseFetcher` interface that has a `getStripeResponse` method, then create a Default implementation of that interface that has all the logic for making a request and handling the response that is passed through in the production code, and a test version that can be passed through in test classes. This feels out of the scope of the current PR, but I wonder if there's another way to test your logic for the time being until we can make this change (I'm happy to take on this work).

Reply:
I might take you up on resolving this on your side 😬

Reply:
Let's leave this as is to unblock the review. I'll look into improving it in a future patch.

Reply:
Ok thanks a tdon